src/parser.rs

```diff
@@ -0,0 +1,435 @@
+//! Recursive-descent parser for semver ranges.
+//!
+//! The parsers is divided into a set of functions, each responsible for parsing a subset of the
+//! grammar.
+//!
+//! # Examples
+//!
+//! ```rust
+//! use semver_parser::parser::Parser;
+//! use semver_parser::range::Op;
+//!
+//! let mut p = Parser::new("^1").expect("a working parser");
```

Comment:
the expect message here is backwards; if this doesn't work out, it would say we had one, but we don't

Reply:
Fixed, thanks.